lib/azure/armrest/exception.rb

```diff
@@ -52,18 +52,16 @@ def to_s
       end
     end
 
-    # A list of predefined exceptions that we wrap around RestClient exceptions.
+    # Redefine the RestClient exception classes as our own.
+    RestClient::Exceptions::EXCEPTIONS_MAP.values.each do |klass|
+      name = klass.to_s.split("::").last
+      if name != 'Exception'
+        name += 'Exception' unless name.include?('Exception')
+        Object.module_eval("class Azure::Armrest::#{name} < Azure::Armrest::ApiException; end")
```

Comment:
@chrisarcand If there's a way, I don't know what it is. I did update the PR to just use class_eval instead of Object.class_eval since we want to make sure we're evaluating in its current context. The tests confirm it.